src/libraries/System.Text.Json/src/System/Text/Json/Node/JsonString.cs

```diff
@@ -130,7 +161,38 @@ public string Value
         ///  <see langword="true"/> if instance was converted successfully;
         ///  otherwise, <see langword="false"/>
         /// </returns>
-        public bool TryGetDateTimeOffset(out DateTimeOffset value) => JsonHelpers.TryParseAsISO(_value.AsSpan(), out value);
+        public bool TryGetDateTimeOffset(out DateTimeOffset value)
+        {
+            ReadOnlySpan<char> span = _value.AsSpan();
+
+            if (!JsonHelpers.IsValidDateTimeOffsetParseLength(span.Length))
+            {
+                value = default;
+                return false;
+            }
+
+            bool isQuoted = span[0] == JsonConstants.Quote && span[span.Length - 1] == JsonConstants.Quote;
```

Comment:
I don't think checking for embedded quotes makes sense here.

We don't do it for other APIs like GetGuid. I would remove this check and trimming.

```C#
var str = new JsonString("\"eb5c8c7d-187a-44e6-8afb-81e854c39457\"");
Console.WriteLine(str.GetGuid()); // Throws System.FormatException : Guid should contain 32 digits with 4 dashes
```

Reply:
So, a quoted string should be taken as an invalid DateTime, i.e., the following code will throw the FormatException. Is that correct?

```csharp
var jsonStr = new JsonString("\"1997-07-16T19:20:30.4555555+14:00\"");
jsonStr.GetDateTime(); // System.FormatException : The JSON value is not in a supported DateTime format.
```

Reply:
Yes.